src/psyclone/tests/psyGen_test.py

```diff
@@ -5513,6 +5532,87 @@ def test_fparser2astprocessor_do_construct_while(f2008_parser):
     assert isinstance(fake_parent.children[0], CodeBlock)
 
 
+# (1/4) fparser2astprocessor::nodes_to_code_block
+def test_fp2astproc_nodes_to_code_block_1(f2008_parser):
+    '''Check that a codeblock that is the first node in the PSyiR has the
+    structure property set to statement.'''
+    from fparser.common.readfortran import FortranStringReader
+    reader = FortranStringReader('''
+        program test
+        do while(a .gt. b)
+            c = c + 1
+        end do
+        end program test
+        ''')
+    prog = f2008_parser(reader)
+    psy = PSyFactory(api="nemo").create(prog)
+    schedule = psy.invokes.invoke_list[0].schedule
+    assert isinstance(schedule[0], CodeBlock)
+    assert schedule[0].structure == CodeBlock.Structure.STATEMENT
+
+
+# (2/4) fparser2astprocessor::nodes_to_code_block
+def test_fp2astproc_nodes_to_code_block_2(f2008_parser):
+    '''Check that a codeblock that is not the first node in the PSyiR and
+    contains a statement has the structure property set to
+    statement.
+
+    '''
+    from fparser.common.readfortran import FortranStringReader
+    reader = FortranStringReader('''
+        program test
+        a=1.0
+        b=0.0
+        do while(a .gt. b)
+            c = c + 1
+        end do
+        end program test
+        ''')
+    prog = f2008_parser(reader)
+    psy = PSyFactory(api="nemo").create(prog)
+    schedule = psy.invokes.invoke_list[0].schedule
+
+    assert isinstance(schedule[2], CodeBlock)
+    assert schedule[2].structure == CodeBlock.Structure.STATEMENT
+
+
+# (3/4) fparser2astprocessor::nodes_to_code_block
+def test_fp2astproc_nodes_to_code_block_3(f2008_parser):
+    '''Check that a codeblock that contains an expression has the
+    structure property set to expression.
+
+    '''
+    from fparser.common.readfortran import FortranStringReader
+    # The string "HELLO" is currently a code block in the PSyIR
+    reader = FortranStringReader('''
+        program test
+        if (a == "HELLO") then
```

Comment:
I think this wont't work if the condition is reversed: `if ("HELLO" == a) then` then it returns a CodeBlock with structure =  STATEMENT

Reply:
Agreed, please see previous comment. I obviously had a serious brain block before.